app/models/ta.rb

```diff
@@ -2,7 +2,9 @@ class Ta < User
 
   CSV_UPLOAD_ORDER = Rails.configuration.ta_csv_upload_order
   SESSION_TIMEOUT = Rails.configuration.ta_session_timeout
-
+  has_one :grader_permission, dependent: :destroy, foreign_key: :user_id
+  after_create :create_grader_permission
```

Comment:
change this to a `before_validation` callback and add a validation for the existence of the associated object:

```
  validates_presence_of :grader_permission
  before_validation :create_grader_permission, on: :create 
```